### DISCUSSION TITLE
[教程] 关于 venus-auth 生成 token 类型的解释

### DISCUSSION THREAD

**Discussion by Fatman13** [filecoin-project/venus#4880] [Tutorials]
[原文](https://filecoinproject.slack.com/archives/C028PCH8L31/p1652358530054209?thread_ts=1652349698.641089&cid=C028PCH8L31) by @Venus-Ops

`token`的`read`、`write`和`admin`三种权限的应用场景：
- read权限一般在配置venus-wallet连接到venus-gateway上使用，以最低权限注册到gateway上；
- write权限一般用于venus-cluster、venus-sealer和venus-maket的solo模式使用；
- admin权限用于链服务组件间访问或者venus-market的pool模式使用

@diwufeiwen 补充：

venus集群中token和user,miner的关系，一个token唯一对应一个user,决定API的访问权限，一个user可以添加多个miner，对应的这个miner访问各组件API的认证token就继承这个user的token，创建token时需要根据对应的场景给予合适的权限。

另，多miner支持功能特性[资料](https://github.com/filecoin-project/venus/discussions/4857)

**Comment by diwufeiwen** [comment]
补充解释：venus集群中token和user,miner的关系, 一个token唯一对应一个user,决定API的访问权限,一个user可以添加多个miner,对应的这个miner访问各组件API的认证token就继承这个user的token,创建token时需要根据对应的场景给予合适的权限

**Comment by diwufeiwen** [comment]
补充资料： https://github.com/filecoin-project/venus/discussions/4857

**Comment by elvin-du** [comment]
还有sign权限？

**Reply by Fatman13** [reply]
这个略老，不知道要不要更新。之前的一个操作就是社区常见问题 FAQ 通过 discussion 的方式收集，更新。这样一个好处是 GH 能比较快速修改，不用 pr 文档，而且支持搜索。文档主要负责教会拉起一个节点。具体生产的一些细节问题都会归总到这儿。基本上 lotus 也是这个操作。

**Reply by elvin-du** [reply]
GitHub经常翻墙很困难，对于国内用户非常不友好。

**Reply by elvin-du** [reply]
sign表示啥意思，可以补充一下

**Reply by Fatman13** [reply]
> GitHub经常翻墙很困难，对于国内用户非常不友好。

文档也一样，也是 github page。😂 科学上网其实还好。大多数技术人员/普通人都会。

> sign表示啥意思，可以补充一下

sign 好像应该是签名权限，但是具体得看看代码。
